api/checks/summaries/completed.go

```diff
@@ -0,0 +1,37 @@
+// Copyright 2018 The WPT Dashboard Project. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package summaries
+
+import "github.com/web-platform-tests/wpt.fyi/shared"
+
+// ResultsComparison is all the fields shared across summaries that
+// involve a result comparison.
+type ResultsComparison struct {
+	MasterRun     shared.TestRun
+	PRRun         shared.TestRun
+	MasterDiffURL string
+	DiffURL       string // URL for the diff-view of the results
+	HostName      string // Host environment name, e.g. "wpt.fyi"
+	HostURL       string // Host environment URL, e.g. "https://wpt.fyi"
+}
+
+// Completed is the struct for completed.md
+type Completed struct {
+	CheckState
+	ResultsComparison
+
+	Results map[string][]int
+	More    int
+}
+
+// GetCheckState returns the info needed to update a check.
```

Comment:
Here and elsewhere. `CheckState` is exported. Is there an interface-related reason that we implement `GetCheckState()`?

Reply:
Purely because (annoyingly) you can't put a field on an interface in Golang. That said, I think I could define the function on CheckState itself and have this only written once.